dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java

```diff
@@ -415,9 +415,18 @@ String xssCheck(String uri, String queryString) throws ServletException {
 	public String getURIFromRequest(HttpServletRequest request)
 			throws UnsupportedEncodingException {
 
-		return (request.getAttribute(CMS_FILTER_URI_OVERRIDE) != null) ? (String) request
+        String requestPath = request.getRequestURI();
+        try {
+            URI requestURI = new URI(requestPath);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'requestURI' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4763071)

Reply:
Done